src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const oldDate = date.split(oldSeparator);
+  const newDate = [];
+  const dateObj = {};
+  let year = 0;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = oldDate[i];
+    }
+
+    dateObj[fromFormat[i]] = oldDate[i];
+  }
+
+  const yearOfCentury = (year < 30) ? 20 : 19;
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    newDate.push(dateObj[toFormat[i]]);
+
+    if (!dateObj.hasOwnProperty(toFormat[i])) {
```

Comment:
It's strange that you have to overwrite element in array after adding, you can wrap `push` in `if` and break loop after element adding. Then there will be no need to wrap other logic of loop in `if` and you'll be able to push modified year to array